client/src/components/WorkspaceNav.jsx

```diff
@@ -23,32 +23,75 @@ const getCollapsedStatus = (pathname) => {
   return collapsedStatus;
 };
 
+const getActiveSubtitle = (pathname) => {
+  let activeSubtitle;
+
+  if (pathname === 'library') {
+    activeSubtitle = 'library';
+  } else if (pathname.indexOf('library') > -1) {
+    activeSubtitle = 'collections';
+  } else if (pathname.indexOf('activity') > -1) {
+    activeSubtitle = 'activity';
+  }
+
+  return activeSubtitle;
+};
+
 export default class WorkspaceNav extends Component {
   constructor() {
     super();
     this.handleShowCreateCollectionModal = this.handleShowCreateCollectionModal.bind(this);
     this.state = {
-      isFloating: false,
+      isManuallyExpanded: false,
+      isManuallyCollapsed: false,
     };
   }
 
-  getOnClick(isCollapsible) {
-    let onClick = null;
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.location.pathname !== this.props.location.pathname) {
+      this.setState({
+        isManuallyExpanded: false,
+        isManuallyCollapsed: false,
```

Comment:
why do we need both `isManuallyExpanded` and `isManuallyCollapsed`? Are they not mutually exclusive?